lib/linter.js

```diff
@@ -151,7 +151,7 @@ const lint = (objectName, object, key, options = {}) => {
             }
             if (rule.pattern) {
                 const { omit, property, split, value } = rule.pattern;
-                const target = object[property]
+                let target = (property === '$key') ? key : object[property];
```

Comment:
I think this can still be a `const` as there is only one assignment from the RHS of that expression?

Reply:
My mistake, fixed and squashed the commits